Source/Core/DolphinQt/Config/GamecubeControllersWidget.cpp

```diff
@@ -25,23 +26,32 @@
 
 #include "InputCommon/GCAdapter.h"
 
-static const std::map<SerialInterface::SIDevices, int> s_gc_types = {
-    {SerialInterface::SIDEVICE_NONE, 0},         {SerialInterface::SIDEVICE_GC_CONTROLLER, 1},
-    {SerialInterface::SIDEVICE_WIIU_ADAPTER, 2}, {SerialInterface::SIDEVICE_GC_STEERING, 3},
-    {SerialInterface::SIDEVICE_DANCEMAT, 4},     {SerialInterface::SIDEVICE_GC_TARUKONGA, 5},
-    {SerialInterface::SIDEVICE_GC_GBA, 6},       {SerialInterface::SIDEVICE_GC_KEYBOARD, 7}};
+static const std::vector<std::pair<SerialInterface::SIDevices, const char*>> s_gc_types = {
+    {SerialInterface::SIDEVICE_NONE, "None"},
+    {SerialInterface::SIDEVICE_GC_CONTROLLER, "Standard Controller"},
+    {SerialInterface::SIDEVICE_WIIU_ADAPTER, "GameCube Adapter for Wii U"},
+    {SerialInterface::SIDEVICE_GC_STEERING, "Steering Wheel"},
+    {SerialInterface::SIDEVICE_DANCEMAT, "Dance Mat"},
+    {SerialInterface::SIDEVICE_GC_TARUKONGA, "DK Bongos"},
+#ifdef HAS_LIBMGBA
+    {SerialInterface::SIDEVICE_GC_GBA_EMULATED, "GBA (Emulated)"},
+#endif
+    {SerialInterface::SIDEVICE_GC_GBA, "GBA (TCP)"},
```

Comment:
I think that this naming can be a little confusing (this makes it sound like TCP options refers to physical GBA as opposed to emulated one which is obviously very wrong). Maybe "GBA (built-in, recommended)"/"GBA (separate emulator)"?

Reply:
I like how it is now, but I can see how it may be confusing.  What if it was like this.

GBA (Integrated) and GBA (External)

I don't know if that fixes what you want though, but it would make it clear what each one did.  I do like how TCP leaves it more consistent with the type of connectivity though.

Reply:
I think your wording is fine: people will look things up online instead of making wrong assumptions should they get confused if we'll go with it.

Of course current wording is *technically correct*, but it isn't very informative to your random causal user.

Reply:
I agree that this wording is confusing. Both are emulated after all, just one internal to Dolphin and one external. Perhaps "Integrated" and "Remote".

Reply:
I'm fine with Integrated and Remote if we want to go that route.  Just let me know so I can re-render things.

Reply:
Integrated plus TCP would be my pick. If we do go with Remote, I wouldn't mind, but please put an i18n comment above the string so that translators don't accidentally think that it means the same kind of "remote" as in "Wii Remote".

Reply:
Just someone @ me if a change is made so I can rerender footage 👍

Reply:
Integrated and TCP would be my pick as well, "Remote" is a bit vague and would need to be changed if a shared memory implementation ever comes up

Reply:
I'll assume we're changing over to Integrated and re-render then.  Seems we have a consensus!